doc/fwutil/fwutil.md

```diff
@@ -462,6 +519,321 @@ Aborted!
+                    "firmware": "bios.bin",
+                    "version": "0ACLH003_02.02.010",
+                },
+                "CPLD": {
+                    "firmware": "cpld.bin",
+                    "utility": "cpldupdate",
+                    "version": "10",
+                },
+                "SSD": {
+                    "firmware": "SSD.bin",
+                    "utility": "ssd_fw_update",
+                    "version": "5",
+                }
+            }
+        }
+    }
+}
+```
+
+##### 2.2.2.4.3.1 Platform Component Firmware Update Utility Interface Requirement
+
+The Utility should support the minimum requirements to perform the fwutil auto-update interface. 
+The minimum requirement is that the component api's get_frimware_version, get_firmware_update_notification(), and auto_update_firmware() needs to be supported by the utility.
+Here are the interface requirements to support them.
+1. {utility} -s(--status) : able to retrieve the current firmware version:
+    - equivalent to the component api's get_firmware_version()
+    - response : installed firmware version
+2. {utility} -n(--notification) : able to provide the update complete action :
+    - equivalent to the component api's get_firmware_update_notification(image_path) 
+    - response : the required action to complete the component installation
+3. {utility} -a(--autoupdate) : able to perform the auto-update action :
+    - equivalent to the component api's auto_update_firmware(image_path, boot_type) 
+    - auto-update interface needs two arguments : image_path and boot_type
+    - response : the result of auto-update as follows
+        ```json
+        "True : updated"
+        "True : scheduled"
+        "True : installed"
+        "False : ${ErrorCode}"
+        ```
+
+**Optional) The utility can be supported for other platform api substitues like `compoenent_update` and `compoenent_install` with |-u(--update)|-i(--install)**
+
+The component utility needs to be called by the FWutil command to perform the firmware auto-update process if it's defined in the `platform_components.json`, otherwise, the platform component api will be called.
+The componet utility path will be pased from the `platform_components.json` and be executed by fwutil.
+Below shows how the utility can be executed for the auto-update interface.
+```bash
+...
+firmware_path = parser[self.__pcp.FIRMWARE_KEY]
+utility_path = parser[self.__pcp.UTILITY_KEY]
+if os.path.isfile(utility_path) and os.access(utility_path, os.X_OK):
+    cmd = "{} -a {} {}".format(
+        utility_path,
+        firmware_path,
+        boot_type
+    )
+    output = subprocess.check_output(cmd, stderr=subprocess.STDOUT, shell=True)
+...
+```
+From the above `platform_components.json` example, if the platform needs the SSD firmware update, then the auto-update can be triggered with following command.
+`$PWD/ssd_fw_update -a $PWD/SSD.bin fast` 
+
+##### 2.2.2.4.3.2 Platform Firmware Update Reboot Handle Plugin
```

Comment:
@sujinmkang please elaborate for a different flows (cold/warm/fast): at which point do you expect to call this plugin?